src/renderer.ts

```diff
@@ -152,6 +158,504 @@ export class DefaultCellRenderer implements ICellRendererFactory {
   }
 }
 
+
+class HeatmapCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: MultiValueColumn, context: IDOMRenderContext): ISVGCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const colorScale = col.getColor();
+    const padding = context.option('rowPadding', 1);
+    return {
+
+      template: `<g class="heatmapcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const rect = d3.select(n).selectAll('rect').data(col.getValue(d.v, d.dataIndex));
+
+        rect.enter().append('rect');
+        rect
+          .attr({
+            y: padding,
+            x: (d, i) => (i * cellDimension),
+            width: cellDimension,
+            height: context.rowHeight(i),
+            fill: (d, i) => colorScale(d)
+          });
+        rect.exit().remove();
+      }
+    };
+  }
+
+
+  createHTML(col: MultiValueColumn, context: IDOMRenderContext): IHTMLCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+
+    return {
+      template: `<div class="heatmapcell" style="top:${padding}px;">
+                                   </div>`,
+      update: (n: HTMLDivElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const div = g.selectAll('div').data(col.getValue(d.v, d.dataIndex));
+        div.enter().append('div');
+        div
+          .style({
+            'width': cellDimension + 'px',
+            'background-color': (d, i) => colorScale(d),
+            'height': context.rowHeight(i) + 'px',
+            'left': (d, i) => (i * cellDimension) + 'px'
+          });
+
+
+      }
+    };
+  }
+
+  createCanvas(col: MultiValueColumn, context: ICanvasRenderContext): ICanvasCellRenderer {
+
+    const cellDimension = col.calculateCellDimension(col.getWidth());
+    const padding = context.option('rowPadding', 1);
+    const colorScale = col.getColor();
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+
+      const data = col.getValue(d.v, d.dataIndex);
+      data.forEach((d: number, j: number) => {
+        const x = (j * cellDimension);
+        ctx.beginPath();
+        ctx.fillStyle = colorScale(d);
+        ctx.fillRect(x, padding, cellDimension, context.rowHeight(j));
```

Comment:
![image](https://cloud.githubusercontent.com/assets/12292333/21742342/de14e582-d4ec-11e6-8e01-d58b47f1e6ea.png)
The i is same for each data of rect. But there are four different rect in heatmap with shifted in x position so i is not correct for this situation. 
![image](https://cloud.githubusercontent.com/assets/12292333/21742350/1b5b1790-d4ed-11e6-91e1-1a8ca23e4d7d.png)
As you can see four rect are overlapped in single one in each row when i is implemented.